libs/libxx/Kconfig

```diff
@@ -87,7 +87,7 @@ endif
 
 config CXX_STANDARD
 	string "Language standard"
-	default "gnu++17"
+	default "gnu++20"
```

Comment:
We shouldn't increase the default to `gnu++20`.
Not even the latest version of GCC has full support for the Language features:
https://gcc.gnu.org/projects/cxx-status.html#cxx20

Even for C++17 is states:
> GCC has almost full support for the previous revision of the C++ standard, which was published in 2017.

Reply:
> We shouldn't increase the default to `gnu++20`. Not even the latest version of GCC has full support for the Language features: https://gcc.gnu.org/projects/cxx-status.html#cxx20
> 
> Even for C++17 the documentation states:
> 
> > GCC has almost full support for the previous revision of the C++ standard, which was published in 2017.

gnu++20 or gnu++17 doesn't mean all features specified by standard are implemented. On the other hand, libcxx 15.0.7 require some c++20 feature which is implemented by new version of gcc and clang to pass the build.

Reply:
> It's a good time to make a default version to c++20. If someone can't switch c++20, he still can change the version by defconfig anyway.

But since the `libcxx` version is not configurable, then users will be restricted to the core language features as NuttX currently does not support an alternative implementation for the standard library.

I am in favor of upgrading `libcxx`, but I am worried about this breaking support for existing users.
Maybe we should extend the `libcxx` support to a set of versions, e.g. both 12.0.0 and 15.0.7.
Does it make sense?

Reply:
> > It's a good time to make a default version to c++20. If someone can't switch c++20, he still can change the version by defconfig anyway.
> 
> But since the `libcxx` version is not configurable, then users will be restricted to the core language features as NuttX currently does not support an alternative implementation for the standard library.
> 

But most new features come from compiler and c++ library, not directly related to NuttX.

> I am in favor of upgrading `libcxx`,

> but I am worried about this breaking support for existing users.

libcxx is very stable and the spec seldom remove the feature from previous version. I don't see the new vession will break the user code. The user can:

1. Either stick in the old release
2. Or revert this change to restore 12.0.0

> Maybe we should extend the `libcxx` support to a set of versions, e.g. both 12.0.0 (for c++17 and older) and 15.0.7 (c++20 and newer). Does it make sense?

12.0.0 isn't a good starting pointer. We need start from a version which doesn't need apply the patch, otherwise it will make the multiple version support become very complex. @g2gps do you know the particular version?

Reply:
> But most new features come from compiler and c++ library, not directly related to NuttX.

The C++ library is related to the underlying OS, so it is related to NuttX.

> If so, why not upgrade?:)

Well, we need to make sure this is not a reckless move. Someone needs to raise these questions.

> libcxx is very stable and the spec seldom remove the feature from previous version. I don't see the new vession will break the user code. The user can:
> 1. Either stick in the old release
> 2. Or revert this change to restore 12.0.0

Both suggestions require patching, which is exactly what I would like to prevent.

Reply:
Release `15.0.0` is the earliest version which has the the changes from the existing 5 patches currently in the NuttX tree applied. This version already doesn't compile with gcu++17 or c++17, without additional patching.

Reply:
I understand that `std=gnu++20` maybe required for `libcxx` to build, but it shouldn't be made as default for the reasons stated above. C++20 support in GCC is still experimental.

`std=c++20` should be restricted to the `libcxx` build.

Reply:
is it good to change the compiler option in the different source? I change the code enable gnu++20 only when LIBCXX is selected. @gustavonihei do you think it's acceptable?